File: Entities/Category.cs
using System.Collections.Generic;

namespace RelationShipManager.Entities
{
    public class Category
    {
        public Category()
        {
            Product = new HashSet<Product>();
        }

        public int IdCategory { get; set; }
        public string Category1 { get; set; }

        public ICollection<Product> Product { get; set; }
    }
}

File: src/app/_models/category.ts
import { Observable } from 'rxjs';
import { Product } from './product';

export class Category {

    idCategory: number;
    category1: string;
    product: Product[];

    constructor(
        idCategory: number,
        name: string,
        products: Product[]
    ) {
        this.idCategory = idCategory;
        this.category1 = name;
        this.product = products;
    }

    public printCategory() {
        console.log(this.idCategory + '   ' + this.category1);
    }
}
